docs/generator.py

```diff
@@ -257,6 +284,152 @@ def _generate_from_template(self, dest_dir, filename):
         self._write_generated_file(dest_dir, output_filename, content)
 
 
+class ResourceModelDocGenerator(ModelDocGenerator):
+
+    """Slightly customised ModelDoc generation class. Required for generation of FTD API Documentation"""
+
+    MODEL_TEMPLATE = 'resource_model.md.j2'
+
+    @staticmethod
+    def _get_index_data(index_name, index_list):
+        return {'index_list': index_list}
+
+
+class ResourceDocGenerator(BaseDocGenerator, OperationDocGenerationMixin):
+    """Generates documentation for the HTTP resources defined in the
+    Swagger specification. Describes available endpoints with their methods,
+    models, query/body/path parameters, etc.
+    Documentation is written using Markdown markup language.
+    """
+
+    OPERATION_TEMPLATE = 'resource_operation.md.j2'
+    OVERVIEW_TEMPLATE = 'resource_overview.md.j2'
+    CONFIG_TEMPLATE = 'resource_config.json.j2'
+
+    def __init__(self, template_dir, template_ctx, api_spec, errors_codes_file):
```

Comment:
Discussed offline that the `Error Codes` section should not be inside `Resources`, so let's remove it from here as well.

Reply:
done